src/neo/Cryptography/ECC/ECDsa.cs

```diff
@@ -26,7 +25,7 @@ public ECDsa(ECPoint publicKey)
         private BigInteger CalculateE(BigInteger n, byte[] message)
         {
             int messageBitLength = message.Length * 8;
-            BigInteger trunc = new BigInteger(message.Reverse().Concat(new byte[1]).ToArray());
+            BigInteger trunc = new BigInteger(message, isUnsigned: true, isBigEndian: true);
```

Comment:
Is this cat 1 zero at lsb correct? It should be cat on top now, no? Its not msb I guess.

Reply:
It cat 1 zero after reverse, so it is little-endian now.

Reply:
Ahhh, I got it now.. its not ToArray, but constructor here. So you're right, the data comes native as BE, and reverse + cat just ensured unsigness. Much simpler now :+1: